### PR TITLE
[#42167] use states for every components from a single global source

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
Introduces states object in the utlis
Used the states objects everywhere where we deal with states.

## Related
https://community.openproject.org/work_packages/42167

Signed-off-by: Parajuli Kiran <kiranparajuli589@gmail.com>

### CODE REVIEW

**Comment by kiranparajuli589** [comment]
need to change the base branch

File: src/components/AdminSettings.vue [hunk 1]
```diff
@@ -65,9 +65,15 @@ export default {
...
-			redirect_uri: window.location.protocol + '//' + window.location.host + generateUrl('/apps/integration_openproject/oauth-redirect'),
 		}
 	},
+	computed: {
+		redirectUri() {
+			return window.location.protocol
+				+ '//' + window.location.host
+				+ generateUrl('/apps/integration_openproject/oauth-redirect')
+		},
+	},
```
Comment (individual-it) [inline src/components/AdminSettings.vue:76]: nothing to do with the states, different PR?

File: src/components/OAuthConnectButton.vue [hunk 2]
```diff
@@ -27,8 +27,11 @@ export default {
 
 	computed: {
 		adminConfigNotOkMessage() {
-			return t('integration_openproject', 'Some OpenProject integration application settings are not working.'
-				+ ' Please contact your Nextcloud administrator.')
+			return t(
+				'integration_openproject',
+				'Some OpenProject integration application settings are not working.'
+					+ ' Please contact your Nextcloud administrator.'
+			)
```
Comment (individual-it) [inline src/components/OAuthConnectButton.vue:34]: nothing to do with the states, different PR?

**Review by individual-it — COMMENTED** [review]
maybe split into different PRs

**Review by individual-it — APPROVED** [review]

**Pull request merged**